code/obj/item/misc_weapons.dm

```diff
@@ -1717,6 +1717,122 @@ obj/item/whetstone
+	icon = 'icons/obj/large/64x32.dmi'
+	icon_state = "halberd"
+	inhand_image_icon = 'icons/mob/inhand/hand_weapons.dmi'
+	item_state = "halberd1"
+
+	w_class = W_CLASS_BULKY
+	two_handed = 1
+	throw_range = 10
+	throwforce = 30 //yeet like spear
+	stamina_crit_chance = 5
+
+	//these combat variables change depending on intent- starts with help intent vars
+	force = 28
+	stamina_damage = 10
+	stamina_cost = 7
+
+
+	hit_type = DAMAGE_CUT
+	flags = FPRINT | TABLEPASS | USEDELAY | ONBACK
+	c_flags = EQUIPPED_WHILE_HELD
+	item_function_flags = USE_INTENT_SWITCH_TRIGGER | USE_SPECIALS_ON_ALL_INTENTS
+
+	var/guard = null
+
+	New()
+		..()
+		BLOCK_SETUP(BLOCK_SWORD)
+
+	setupProperties()
+		. = ..()
+		setProperty("deflection", 60)
+		setProperty("block", 40)
+
+
+	intent_switch_trigger(mob/user as mob)
+		if(guard != user.a_intent)
+			change_guard(user,user.a_intent)
+
+	proc/change_guard(var/mob/user,var/intent) //heavily modified kendo code
+		guard = intent
+		switch(guard)
+			if("help") //light swing with the axe
+				force = 24
+				stamina_damage = 10
+				stamina_cost = 7
+				item_state = "halberd2"
+				icon_state = "halberd"
+				hit_type = DAMAGE_CUT
+				src.click_delay = COMBAT_CLICK_DELAY * 0.75
+				hitsound =  'sound/impact_sounds/Blade_Small_Bloody.ogg'
+				src.setItemSpecial(/datum/item_special/simple)
+				boutput(user, "<span class='notice'>You will now make light swings with the axe!</span>")
+			if("disarm") //thrust with the pointy end
+				force = 15
+				stamina_damage = 10
+				stamina_cost = 8
+				item_state = "halberd2"
+				icon_state = "halberd"
+				hit_type = DAMAGE_STAB
+				src.click_delay = COMBAT_CLICK_DELAY * 0.60
+				hitsound = 'sound/impact_sounds/Flesh_Stab_1.ogg'
+				src.setItemSpecial(/datum/item_special/rangestab)
+				boutput(user, "<span class='notice'>You will thrust with the tip!</span>")
+
+			if("grab") //attack with the spur on the back
+				force = 10
+				stamina_damage = 40
+				stamina_cost = 20
+				item_state = "halberd1"
+				icon_state = "halberd2"
+				hit_type = DAMAGE_STAB
+				src.click_delay = COMBAT_CLICK_DELAY
+				hitsound ='sound/impact_sounds/coconut_break.ogg' //it's a good hitsound when you ignore the name
+				src.setItemSpecial(/datum/item_special/simple)
+				boutput(user, "<span class='notice'>You will now make dehabilitating swings with the spur!</span>")
+
+			if("harm") //wide, tiring swings with the axe
+				force = 35
+				stamina_damage = 20
+				stamina_cost = 35
+				item_state = "halberd1"
+				icon_state = "halberd"
+				hit_type = DAMAGE_CUT
+				src.click_delay = COMBAT_CLICK_DELAY * 1.25
+				hitsound =  'sound/impact_sounds/Blade_Small_Bloody.ogg'
+				src.setItemSpecial(/datum/item_special/swipe)
+				boutput(user, "<span class='notice'>You will now make heavy swings with the axe!</span>")
+
+
+		user.update_inhands()
+		src.buildTooltipContent()
```

Comment:
```suggestion
		src.tooltip_rebuild = TRUE
```
This updates the tooltip 'lazily', so we don't actually update the text unless someone tries to check it.